src/ipv4.c

```diff
@@ -122,6 +122,25 @@ static int ipv4_get_route(struct rtentry *route)
 	uint32_t rtdest, rtmask, rtgtw;
 	int rtfound = 0;
 
+	/* initialize the buffer with zeroes, aiming to address the
+	 * coverity issue "tainted_skalar passed to a tainted sink"
+	 *
+	 * Later on, the routing table is read into this buffer using
+	 * read() and therefore the content of the buffer is considered
+	 * tainted. strtok_r internally uses it in a loop boundary.
+	 * The theoretical problem is that the loop could iterate forever,
+	 * if the buffer contains a huge string which doesn't contain
+	 * the token character, which we are parsing for.
+	 *
+	 * We can declare this as a false positive, because
+	 * - the routing table is to some extent trusted input,
+	 * - it's not that large,
+	 * - and the loop in strtok_r increments the pointer in each
+	 *   interation until it reaches the area where we have ensured
+	 *   that there is a delimiting '\0' character by proper
+	 *   initialization. We ensure this also when growing the buffer.
+	 */
```

Comment:
Agreed.

That said perhaps we should use a 3rd party parser generator (some _yacc_ alternative?) instead of parsing everything manually. It would probably avoid such warnings. In the same way HTML parsing could use a 3rd party library as already suggested. Indeed I think most Coverity issues were found in parsing code. That would be for an upcoming release though...

Reply:
yes, that would be a bigger change. We could add this parsing topic to #347 or open a separate issue so that we don't forget it...